src/unix/udp.c

```diff
@@ -180,6 +184,36 @@ static int uv__udp_recvmmsg(uv_udp_t* handle, uv_buf_t* buf) {
   while (nread == -1 && errno == EINTR);
 
   if (nread < 1) {
+    if (nread == -1) {
+      #if defined(__linux__) && defined(MSG_ERRQUEUE)
+      if (handle->flags & UV_HANDLE_UDP_RECVERR) {
+        struct msghdr h;
+        struct sockaddr_storage peer;
+        char control_buffer[256];
+        memset(&h, 0, sizeof(h));
+        memset(&peer, 0, sizeof(peer));
+        h.msg_name = &peer;
+        h.msg_namelen = sizeof(peer);
+        h.msg_iov = (void*) &buf;
+        h.msg_iovlen = 1;
+        h.msg_control = control_buffer;
+        h.msg_controllen = 256;
+        h.msg_flags = 0;
+        int ret = recvmsg(handle->io_watcher.fd, &h, MSG_ERRQUEUE);
+        if (ret >= 0 && h.msg_flags & MSG_ERRQUEUE) {
+          struct cmsghdr *cmsg = CMSG_FIRSTHDR(&h);
+          if (cmsg && cmsg->cmsg_level == SOL_IP && cmsg->cmsg_type == IP_RECVERR) {
+            struct sock_extended_err *e = (struct sock_extended_err *)CMSG_DATA(cmsg);
+            flags = UV_UDP_RECVERR;
+            if (h.msg_flags & MSG_TRUNC)
+              flags |= UV_UDP_PARTIAL;
+            handle->recv_cb(handle, UV__ERR(e->ee_errno), buf, (const struct sockaddr*) &peer, flags);
+          }
+          return nread;
+        }
+      }
+      #endif
```

Comment:
I don't think this is the best place to handle it (and the fact that you need to sort-of duplicate this logic below underlines that) but what you could do instead, is check `revents & POLLERR` in `uv__udp_io`.

Maybe even better is to pass `revents` to `uv__udp_recvmmsg` so it knows when to pass MSG_ERRQUEUE to recvmsg().

That said, what happens when there's an error and `uv_udp_recv_stop` is called? `uv__udp_recvmmsg` won't get called in that case but libuv can still be polling for EPOLLOUT, so I expect the error is going to stay in the queue, resulting in the same busy loop as before.

Reply:
Oh.I have update the code.